src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -105,6 +112,10 @@ public MaxwellKafkaProducer(MaxwellContext context, Properties kafkaProperties,
 
 	@Override
 	public void push(RowMap r) throws Exception {
+		if (r instanceof DDLMap && !outputDDL) {
+			return;
+		}
```

Comment:
the loose contract we put in place for HeartbeatRowMap should be used here:  if a row shouldn't be output, it should return `null` from toJSON; this will make all the producers honor the `outputDDL` flag rather (implicitly) rather than having to modify each producer.

more succintly: should put `outputDDL` into `MaxwellOutputConfig`, and have the `DDLRowMap` return `null` from `toJSON` if we shouldn't output it.

(that should fix your failing tests, too)